### PR TITLE
rune/libenclave/skeleton: fix return value error

### PR DESCRIPTION
**Pull request by haosanzi**
Signed-off-by: Shirong Hao <shirong@linux.alibaba.com>

### CODE REVIEW

**Review by hustliyilin — COMMENTED** [review]
The `skeleton` appears more than once in the commit title.

File: rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c [hunk 1]
```diff
@@ -841,18 +841,12 @@ int __pal_kill(int pid, int sig)
 
 int __pal_destroy(void)
 {
-	FILE *fp = fdopen(pal_stdio.stderr, "w");
-	if (!fp)
-		return -1;
-
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c:847]: Need to check the validity of pal_stdio and then decide to use pal_stdio.stderr or stderr.

File: rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c [hunk 2]
```diff
@@ -697,7 +698,7 @@ int __pal_exec(char *path, char *argv[], pal_stdio_fds *stdio, int *exit_code)
 	}
 
 	fprintf(fp, "Enclave runtime skeleton initialization succeeded\n");
-	fclose(fp);
+	fflush(fp);
```
Comment (YangLiang3) [inline rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c:701]: why not fclose(fp)? There is one flush operation in fclose(fd)

Comment (haosanzi) [reply rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c:701]: If fclose(fp), then in `__pal_destroy` function, `fdopen(pal_stdio.stderr)` will returns `NULL` with the `Bad file descriptor `

**Review by jiazhang0 — APPROVED** [review]

File: rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c [hunk 3]
```diff
@@ -827,12 +827,22 @@ int __pal_get_local_report(void *targetinfo, int targetinfo_len,
...
+		fp = fdopen(pal_stdio.stderr, "w");
+		if (!fp)
+			return -1;
+	}
+
 	if (!initialized) {
-		fprintf(stderr,
-			"Enclave runtime skeleton uninitialized yet!\n");
+		fprintf(fp, "Enclave runtime skeleton uninitialized yet!\n");
+		fclose(fp);
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c:840]: You should do the same thing as __pal_exec(). Use fflush() here.

File: rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c [hunk 4]
```diff
@@ -827,12 +827,22 @@ int __pal_get_local_report(void *targetinfo, int targetinfo_len,
...
 		return -1;
 	}
 
+	fclose(fp);
```
Comment (jiazhang0) [inline rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c:844]: You should not do this. Nor stderr and pal_stdio.stderr.

**Review by jiazhang0 — APPROVED** [review]

**Pull request merged**